src/Umbraco.Forms.Integrations.Crm.ActiveCampaign/Client/src/property-editor/contact-mapping/contact-mapping-property-editor.element.ts

```diff
@@ -0,0 +1,198 @@
+import { html, customElement, property, state, css, map, when } from '@umbraco-cms/backoffice/external/lit';
+import type { UmbPropertyEditorUiElement } from '@umbraco-cms/backoffice/extension-registry';
+import { UmbLitElement } from '@umbraco-cms/backoffice/lit-element';
+import { ACTIVECAMPAIGN_CONTEXT_TOKEN } from '@umbraco-integrations/activecampaign/context';
+import { ContactFieldSettings, Field } from '@umbraco-integrations/activecampaign/generated';
+import { UUISelectEvent } from '@umbraco-cms/backoffice/external/uui';
+import { ContactMappingValue } from '../../models/activecampaign.model';
+
+const elementName = "contact-mapping-property-editor";
+
+@customElement(elementName)
+export class ContactMappingPropertyUiElement extends UmbLitElement implements UmbPropertyEditorUiElement {
+  #activeCampaignContext!: typeof ACTIVECAMPAIGN_CONTEXT_TOKEN.TYPE;
+
+  @property({ type: String })
+  public value = "";
+
+  @state()
+  public contactMappingArray : Array<ContactMappingValue> = [];
+
+  @state()
+  private selectedContactField: string = "";
+
+  @state()
+  private selectedFormField: string = "";
+
+  @state()
+  private contactFields: Array<ContactFieldSettings> | undefined = [];
+
+  @state()
+  private formdFields: Array<Field> | undefined = [];
+
+  constructor() {
+    super();
+    this.consumeContext(ACTIVECAMPAIGN_CONTEXT_TOKEN, (context) => {
+        if (!context) return;
+        this.#activeCampaignContext = context;
+    });
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+
+    if(this.value){
+      this.contactMappingArray = JSON.parse(this.value);
+    }
+
+    await this.#getContactFields();
+    await this.#getFormFields();
+  }
+
+  async #getContactFields(){
+    var result = await this.#activeCampaignContext.getContactFields();
+    if (!result) return;
+
+    this.contactFields = result.data;
+  }
+
+  async #getFormFields(){
+    var formId = window.location.pathname.split("/")[7]; //Get the formid based on current url.
```

Comment:
Of course this approach works, but it looks ... not nice ... Any other ideas about getting current formId are welcome.